filament/backend/src/vulkan/VulkanFboCache.h

```diff
@@ -39,25 +46,36 @@ class VulkanFboCache {
     // RenderPassKey is a small POD representing the immutable state that is used to construct
     // a VkRenderPass. It is hashed and used as a lookup key.
     struct alignas(8) RenderPassKey {
-        VkImageLayout colorLayout[MRT::MAX_SUPPORTED_RENDER_TARGET_COUNT];  // 32 bytes
+        // For each target, we need to know three image layouts: the layout BEFORE the pass, the
+        // layout DURING the pass, and the layout AFTER the pass. Here are the rules:
+        // - For depth, we explicitly specify all three layouts.
+        // - Color targets have their initial image layout specified with a bitmask.
+        // - For each color target, the pre-existing layout is either UNDEFINED (0) or GENERAL (1).
+        // - The render pass and final images layout for color buffers is always GENERAL.
+        uint8_t initialColorLayoutMask;          // 1 byte
+        VulkanDepthLayout initialDepthLayout;    // 1 byte
+        VulkanDepthLayout renderPassDepthLayout; // 1 byte
+        VulkanDepthLayout finalDepthLayout;      // 1 byte (for now this is always GENERAL)
```

Comment:
You could probably use a bitfield (`:2`) here and "compress" those 3 bytes into one?